### PR TITLE
AttributeController Acceptance Test

### PR DESCRIPTION
**Pull request by pawel-krupinski-guardian**
Test that:
- Starts the service programmatically (inside a Play Server)
- Mocks responses of external http calls (like identity service)
- Interacts with membership-data-api service by calling its REST endpoint(s) and asserting on the response

To make this test possible I replaced many references to configuration.Config fields, which is a singleton that loads config (and therefore doesn't allow overriding)  with usage of an instance of Typesafe Config (that comes from Play configuration).

TODO (in the future): Use embedded Postgres and Dynamo instead of mocks.

### CODE REVIEW

File: membership-attribute-service/test/acceptance/AttributeControllerAcceptanceTest.scala [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+      (json \ "contentAccess" \ "paidMember").as[Boolean] shouldEqual false
+      (json \ "contentAccess" \ "recurringContributor").as[Boolean] shouldEqual false
+      (json \ "contentAccess" \ "supporterPlus").as[Boolean] shouldEqual false
+      (json \ "contentAccess" \ "digitalPack").as[Boolean] shouldEqual true
+      (json \ "contentAccess" \ "guardianWeeklySubscriber").as[Boolean] shouldEqual false
+      (json \ "contentAccess" \ "guardianPatron").as[Boolean] shouldEqual false
+    }
+  }
+
+  val identityResponse =
```
Comment (rupertbates) [inline membership-attribute-service/test/acceptance/AttributeControllerAcceptanceTest.scala:106]: Do we get this full reponse from Identity currently?

File: project/Dependencies.scala [hunk 2]
```diff
@@ -44,6 +46,8 @@ object Dependencies {
     "com.okta.jwt" % "okta-jwt-verifier" % oktaJwtVerifierVersion,
     "com.okta.jwt" % "okta-jwt-verifier-impl" % oktaJwtVerifierVersion,
   )
+  val unirest = "com.konghq" % "unirest-java" % "4.0.0-RC2" % Test
+  val okHttp = "com.squareup.okhttp3" % "okhttp" % "5.0.0-alpha.10"
```
Comment (rupertbates) [inline project/Dependencies.scala:50]: Do we need to use an alpha version of okhttp?

File: membership-attribute-service/test/acceptance/AttributeControllerAcceptanceTest.scala (hunk 1 above)
Comment (pawel-krupinski-guardian) [reply membership-attribute-service/test/acceptance/AttributeControllerAcceptanceTest.scala:106]: Yeah I coped it from a real response.

File: project/Dependencies.scala (hunk 2 above)
Comment (pawel-krupinski-guardian) [reply project/Dependencies.scala:50]: It's to resolve a dependency conflict.
It's a transitive dependency and I think sbt might be already using that version.
Will double check.

Comment (pawel-krupinski-guardian) [reply project/Dependencies.scala]: Ok, explicit dependency (override) wasn't needed, just an exclusion.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -44,6 +46,8 @@ object Dependencies {
     "com.okta.jwt" % "okta-jwt-verifier" % oktaJwtVerifierVersion,
     "com.okta.jwt" % "okta-jwt-verifier-impl" % oktaJwtVerifierVersion,
   )
+  val unirest = "com.konghq" % "unirest-java" % "4.0.0-RC2" % Test
+//  val okHttp = "com.squareup.okhttp3" % "okhttp" % "5.0.0-alpha.10"
```
Comment (rupertbates) [inline project/Dependencies.scala:50]: Can you remove the comment please?

**Review by rupertbates — APPROVED** [review]
Just one little nitpick, but good work! 👍

File: project/Dependencies.scala (hunk 3 above)
Comment (pawel-krupinski-guardian) [reply project/Dependencies.scala]: Fixed now

**Pull request merged**